linodecli/configuration/__init__.py

```diff
@@ -275,7 +199,53 @@ def plugin_get_value(self, key):
 
         return self.config.get(username, full_key)
 
-    def write_config(self, silent=False):
+    # TODO: this is more of an argparsing function than it is a config function
+    # might be better to move this to argparsing during refactor and just have
+    # configuration return defaults or keys or something
+    def update(self, namespace, allowed_defaults):
+        """
+        This updates a Namespace (as returned by ArgumentParser) with config values
+        if they aren't present in the Namespace already.
+        """
+        if self.used_env_token and self.config is None:
+            return None
+        username = self.username or self.default_username()
+        if (not self.config.has_option(username, "token")
+            and not os.environ.get(ENV_TOKEN_NAME, None)):
+            print(f"User {username} is not configured.")
+            sys.exit(1)
+        if (not self.config.has_section(username)
+            or allowed_defaults is None):
+            return namespace
+
+        warn_dict = {}
+        ns_dict = vars(namespace)
+        for key in allowed_defaults:
+            if key not in ns_dict:
+                continue
+            if ns_dict[key] is not None:
+                continue
+            # plugins set config options that start with 'plugin-'
+            # these don't get included in the updated namespace
+            if key.startswith("plugin-"):
+                continue
+            if self.config.has_option(username, key):
+                value = self.config.get(username, key)
+            else:
+                value = allowed_defaults[key]
+            if key == "authorized_users":
+                ns_dict[key] = [value]
+                warn_dict[key] = [value]
+            else:
+                ns_dict[key] = value
+                warn_dict[key] = value
```

Comment:
Wonder if we could break this key/value normalization logic into a helper function

Reply:
i feel since it relies so heavily on self.config and effectively returns 2 dictionaries its fine as it is now but if we refactor it into argparse i think we can clean it up some